src/main/java/io/openliberty/tools/common/plugins/util/InstallFeatureUtil.java

```diff
@@ -458,7 +489,21 @@ private File downloadEsaArtifact(String mavenCoordinates) throws PluginExecution
         String groupId = mavenCoordinateArray[0];
         String artifactId = mavenCoordinateArray[1];
         String version = mavenCoordinateArray[2];
-        return downloadArtifact(groupId, artifactId, "esa", version);
+        File downloadedEsa = downloadArtifact(groupId, artifactId, "esa", version);
+        if(this.verifyOption != VerifyOption.skip) {
+            //download signature file for this esa
+            try {
+        	downloadSignature(downloadedEsa, groupId, artifactId, "esa.asc", version);
+            }catch(PluginExecutionException e) {
+        	if(this.verifyOption == VerifyOption.all) {
+        	    throw e;
```

Comment:
`enforce` verifies all Liberty features. 
`all` verifies both Liberty and user featuers.
`warn` verifies both Liberty and user features, but if there's signature verification failure, it will result in warning message. 

At this point, we don't know if the download failed for Liberty and or user feature, so we don't throw exception for `enforce`.

Reply:
In that case, probably would be good to add some comments to the code explaining this.